src/components/editor/Store/LiveSpecsHydrator.tsx

```diff
@@ -29,15 +27,12 @@ function LiveSpecsHydrator({
     const setSpecs = useEditorStore_setSpecs({ localScope: localZustandScope });
     const setId = useEditorStore_setId({ localScope: localZustandScope });
 
-    useEffect(() => {
-        setId(lastPubId);
-    }, [lastPubId, setId]);
-
     useEffect(() => {
         if (hasLength(publicationSpecs)) {
             setSpecs(publicationSpecs);
+            setId(publicationSpecs[0].last_pub_id);
```

Comment:
If we just set the `id` in the hydrator then we do not have to pass it around as much

Reply:
Yeah - the concept of `id` has caused a few issues while working on this but yeah... didn't want to boil the ocean on this one.